test/integration/puppetlabs/general_puppet/general_puppet_int_test.clj

```diff
@@ -76,50 +82,68 @@
+          (is (= 200 (:status response)))
+          (is (= "test foobar modules/foo/files/bar.txt\n" (:body response)))))
 
-        (let [error-message "Error: A /static_file_content request requires an environment, a code-id, and a file-path"]
-          (testing "the /static_file_content endpoint returns an error if code_id is not provided"
-            (let [response (http-client/get "https://localhost:8140/puppet/v3/static_file_content/foo/bar/?environment=test"
-                                            (assoc testutils/ssl-request-options
-                                              :as :stream))]
-              (is (= 400 (:status response)))
-              (is (= error-message (slurp (:body response))))))
-          (testing "the /static_file_content endpoint returns an error if environment is not provided"
-            (let [response (http-client/get "https://localhost:8140/puppet/v3/static_file_content/foo/bar/?code_id=foobar"
-                                            (assoc testutils/ssl-request-options
-                                              :as :stream))]
-              (is (= 400 (:status response)))
-              (is (= error-message (slurp (:body response))))))
-          (testing "the /static_file_content endpoint returns an error if file-path is not provided"
-            (let [response (http-client/get "https://localhost:8140/puppet/v3/static_file_content/?code_id=foobar&environment=test"
-                                            (assoc testutils/ssl-request-options
-                                              :as :stream))]
-              (is (= 400 (:status response)))
-              (is (= error-message (slurp (:body response)))))))))
+      (let [error-message "Error: A /static_file_content request requires an environment, a code-id, and a file-path"]
+        (testing "the /static_file_content endpoint returns an error if code_id is not provided"
+          (let [response (get-static-file-content "modules/foo/files/bar.txt?environment=test")]
+            (is (= 400 (:status response)))
+            (is (= error-message (:body response)))))
+        (testing "the /static_file_content endpoint returns an error if environment is not provided"
+          (let [response (get-static-file-content "modules/foo/files/bar.txt?code_id=foobar")]
+            (is (= 400 (:status response)))
+            (is (= error-message (:body response)))))
+        (testing "the /static_file_content endpoint returns an error if file-path is not provided"
+          (let [response (get-static-file-content "?code_id=foobar&environment=test")]
+            (is (= 400 (:status response)))
+            (is (= error-message (:body response))))))
+      (let [error-message "Request Denied: A /static_file_content request must be a file within the files directory of a module or a module in an environment"
+            valid-paths ["modules/foo/files/bar.txt"
+                         "modules/foo/files/bar/"
```

Comment:
So a well behaved agent won't ask for a directory, but I don't think we can reliably know if the path in a request is a directory or not. We could reject requests with the path ending in `/`, but `modules/foo/files/bar` would still be valid and could be a directory.